### PR TITLE
[CELEBORN-1171] Add UT for LifecycleManager's async setup endpoints

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
as title

### Why are the changes needed?
as title

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Passes GA

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @cxzl25 @SteNicholas

**Review by FMX — APPROVED** [review]
LGTM.

File: client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala [hunk 1]
```diff
@@ -343,6 +343,57 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
       handleReportShuffleFetchFailure(context, appShuffleId, shuffleId)
   }
 
+  def setupEndpoints(
+      slots: WorkerResource,
+      shuffleId: Int,
+      connectFailedWorkers: ShuffleFailedWorkers): Unit = {
+    val futures = new util.LinkedList[(Future[RpcEndpointRef], WorkerInfo)]()
+    slots.asScala foreach { case (workerInfo, _) =>
```
Comment (cfmcgrady) [inline client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala:351]:
```suggestion
    slots.asScala.foreach { case (workerInfo, _) =>
```

https://github.com/databricks/scala-style-guide?tab=readme-ov-file#infix-methods

File: tests/spark-it/src/test/scala/org/apache/celeborn/tests/client/LifecycleManagerSetupEndpointSuite.scala [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      "celeborn.master.port" -> masterPort.toString)
+    val workerConf = Map(
+      "celeborn.master.endpoints" -> s"localhost:$masterPort")
+    setUpMiniCluster(masterConf, workerConf)
+  }
+
+  test("test setup endpoints with all workers good") {
+    val lifecycleManager: LifecycleManager = new LifecycleManager(APP, celebornConf)
+    val ids = new util.ArrayList[Integer](100)
+    0 until 100 foreach { ids.add(_) }
```
Comment (cfmcgrady) [inline tests/spark-it/src/test/scala/org/apache/celeborn/tests/client/LifecycleManagerSetupEndpointSuite.scala:51]: ditto.

File: tests/spark-it/src/test/scala/org/apache/celeborn/tests/client/LifecycleManagerSetupEndpointSuite.scala [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    lifecycleManager.setupEndpoints(res.workerResource, 0, connectFailedWorkers)
+    assert(connectFailedWorkers.isEmpty)
+
+    lifecycleManager.stop()
+  }
+
+  test("test setup endpoints with one worker down") {
+    val lifecycleManager: LifecycleManager = new LifecycleManager(APP, celebornConf)
+    val ids = new util.ArrayList[Integer](100)
+    0 until 100 foreach {
```
Comment (cfmcgrady) [inline tests/spark-it/src/test/scala/org/apache/celeborn/tests/client/LifecycleManagerSetupEndpointSuite.scala:66]: ditto.

File: tests/spark-it/src/test/scala/org/apache/celeborn/tests/client/LifecycleManagerSetupEndpointSuite.scala [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    val connectFailedWorkers = new ShuffleFailedWorkers()
+    lifecycleManager.setupEndpoints(res.workerResource, 0, connectFailedWorkers)
+    assert(connectFailedWorkers.size() == 1)
+    assert(connectFailedWorkers.keySet().asScala.head == firstWorker.workerInfo)
+
+    lifecycleManager.stop()
+  }
+
+  override def afterAll(): Unit = {
+    logInfo("all test complete , stop celeborn mini cluster")
```
Comment (cfmcgrady) [inline tests/spark-it/src/test/scala/org/apache/celeborn/tests/client/LifecycleManagerSetupEndpointSuite.scala:86]: nit
```suggestion
    logInfo("all test complete, stop celeborn mini cluster")
```

**Pull request closed without merging**